File: src/data/courses/dsa/arrays-and-hashing/advanced-hashing/first-missing-positive/code/java/FirstMissingPositiveSol1.java
class Solution {
    // Hash Set: O(n) time, O(n) space
    public int firstMissingPositive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int n : nums) set.add(n);
        for (int i = 1; i <= nums.length + 1; i++)
            if (!set.contains(i)) return i;
        return nums.length + 1;
    }
}

File: src/data/courses/dsa/arrays-and-hashing/advanced-hashing/first-missing-positive/code/cpp/first_missing_positive_sol1.cpp
class Solution {
public:
    // Hash Set: O(n) time, O(n) space
    int firstMissingPositive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        for (int i = 1; i <= nums.size() + 1; i++) {
            if (!numSet.count(i)) return i;  // First missing
        }
        return nums.size() + 1;
    }
};